modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/NullSafeValueResolverWrapper.java

```diff
@@ -109,8 +112,9 @@ public void visitObject(ObjectType objectType) {
           }
 
           Optional<String> defaultValue = getDefaultValue(objectField);
-          if (defaultValue.isPresent()) {
-            resolver = getFieldDefaultValueValueResolver(objectField, muleContext);
+          if (defaultValue.isPresent() || field.getAnnotation(DefaultEncoding.class) != null) {
+            resolver = getDefaultValueResolver(field.getAnnotation(DefaultEncoding.class) != null, muleContext,
```

Comment:
i am willing to accept this for the time being in order to move forward, but this is a bad solution. We need to have a new component which allows us to pile up all this types of resolvers and apply them anywhere. 

We can't keep adding these ifs everywhere. All of that needs to be centralized. File a jira for that and add a todo